From b9802b59a72c8ed7b6f561d6037bbc17bd6b07cc Mon Sep 17 00:00:00 2001
From: Stella Laurenzo <stellaraccident@gmail.com>
Date: Fri, 11 Apr 2025 17:53:04 -0700
Subject: [PATCH 5/5] Replace refs to GTEST_BOTH_LIBRARIES with individual
 libraries.

* This variable is actually only provided by the finder module and is obsolete (the upstream package does not provide it).
* Modern cmake norms just calls for using the libraries themselves.
---
 clients/benchmarks/CMakeLists.txt | 4 ++--
 clients/gtest/CMakeLists.txt      | 6 ++----
 2 files changed, 4 insertions(+), 6 deletions(-)

diff --git a/clients/benchmarks/CMakeLists.txt b/clients/benchmarks/CMakeLists.txt
index 0a56a91..3babfa6 100644
--- a/clients/benchmarks/CMakeLists.txt
+++ b/clients/benchmarks/CMakeLists.txt
@@ -82,8 +82,8 @@ target_include_directories( hipblas_v2-bench
     $<BUILD_INTERFACE:${FLAME_INCLUDE_DIR}>
 )
 
-target_link_libraries( hipblas-bench PRIVATE roc::hipblas ${GTEST_BOTH_LIBRARIES} )
-target_link_libraries( hipblas_v2-bench PRIVATE roc::hipblas ${GTEST_BOTH_LIBRARIES} )
+target_link_libraries( hipblas-bench PRIVATE roc::hipblas GTest::gtest GTest::gtest_main )
+target_link_libraries( hipblas_v2-bench PRIVATE roc::hipblas GTest::gtest GTest::gtest_main )
 
 if (NOT WIN32)
     target_link_libraries( hipblas-bench PRIVATE hipblas_fortran_client )
diff --git a/clients/gtest/CMakeLists.txt b/clients/gtest/CMakeLists.txt
index eae5389..4b5a2c6 100644
--- a/clients/gtest/CMakeLists.txt
+++ b/clients/gtest/CMakeLists.txt
@@ -142,7 +142,6 @@ target_include_directories( hipblas-test
     $<BUILD_INTERFACE:${BLAS_INCLUDE_DIR}>
     $<BUILD_INTERFACE:${BLIS_INCLUDE_DIR}>
     $<BUILD_INTERFACE:${FLAME_INCLUDE_DIR}>
-    $<BUILD_INTERFACE:${GTEST_INCLUDE_DIRS}>
     ${ROCM_PATH}/include
 )
 target_include_directories( hipblas_v2-test
@@ -152,12 +151,11 @@ target_include_directories( hipblas_v2-test
     $<BUILD_INTERFACE:${BLAS_INCLUDE_DIR}>
     $<BUILD_INTERFACE:${BLIS_INCLUDE_DIR}>
     $<BUILD_INTERFACE:${FLAME_INCLUDE_DIR}>
-    $<BUILD_INTERFACE:${GTEST_INCLUDE_DIRS}>
     ${ROCM_PATH}/include
 )
 
-target_link_libraries( hipblas-test PRIVATE roc::hipblas ${GTEST_BOTH_LIBRARIES} )
-target_link_libraries( hipblas_v2-test PRIVATE roc::hipblas ${GTEST_BOTH_LIBRARIES} )
+target_link_libraries( hipblas-test PRIVATE roc::hipblas GTest::gtest GTest::gtest_main )
+target_link_libraries( hipblas_v2-test PRIVATE roc::hipblas GTest::gtest GTest::gtest_main )
 
 if (NOT WIN32)
     target_link_libraries( hipblas-test PRIVATE hipblas_fortran_client )
-- 
2.43.0

